### PR TITLE
Force driver to use Vert.x event loop

### LINKED ISSUE #21: Integrate Vert.x and driver's event loop

### PR DESCRIPTION
**Pull request by Sammers21**
fixes #21

### CODE REVIEW

**Comment by Sammers21** [comment]
@tsegismont , can you review, please?

File: src/main/java/io/vertx/cassandra/impl/CassandraClientImpl.java [hunk 1]
```diff
@@ -70,7 +74,35 @@ public CassandraClient connect(String keyspace, Handler<AsyncResult<Void>> conne
...
+        public void onClusterClose(EventLoopGroup eventLoopGroup) {
+          // it is important to not do anything here
+          // because the default behaviour is to shutdown the Vert.x event loop group
+        }
+
+        @Override
+        public Timer timer(ThreadFactory threadFactory) {
+          // we don't wan't to use the Vert.x event loop group here
+          // since the timer thread is got blocked between events(via sleep call)
+          return super.timer(threadFactory);
```
Comment (tsegismont) [inline src/main/java/io/vertx/cassandra/impl/CassandraClientImpl.java]: I believe you could keep only the two methods above.

File: src/main/java/io/vertx/cassandra/impl/CassandraClientImpl.java [hunk 2]
```diff
@@ -70,7 +74,35 @@ public CassandraClient connect(String keyspace, Handler<AsyncResult<Void>> conne
       }
     }
 
-    Cluster build = builder.withPort(options.port()).build();
+    Cluster build = builder.withNettyOptions(
+      new NettyOptions() {
```
Comment (tsegismont) [inline src/main/java/io/vertx/cassandra/impl/CassandraClientImpl.java]: Can you move this to an inner class which just takes the `vertx` instance as constructor argument?

**Review by tsegismont — CHANGES_REQUESTED** [review]
Looks good to me. Any other comments @vietj?

File: src/main/java/io/vertx/cassandra/impl/CassandraClientImpl.java (hunk 2 above)
Comment (Sammers21) [reply src/main/java/io/vertx/cassandra/impl/CassandraClientImpl.java]: Yes, just done

File: src/main/java/io/vertx/cassandra/impl/CassandraClientImpl.java (hunk 1 above)
Comment (Sammers21) [reply src/main/java/io/vertx/cassandra/impl/CassandraClientImpl.java]: Agree. Done

**Comment by vietj** [comment]
we should check if Cassandra driver allows the creation of channels, in this case we should use Vert.x Transport class to create them, so the driver can work with native IOs.

**Comment by vietj** [comment]
otherwise we should mention that we only support JDK IO

**Comment by tsegismont** [comment]
@Sammers21 any updates about this one?

**Comment by Sammers21** [comment]
@vietj, @tsegismont, it seems that driver only allows us to specify channel class.

**Comment by Sammers21** [comment]
https://github.com/datastax/java-driver/blob/3.x/driver-core/src/main/java/com/datastax/driver/core/NettyUtil.java#L143

It also looks as if the driver running on Linux(with the specific kernel version), it will use epoll by default.

**Comment by Sammers21** [comment]
@vietj , ping

**Comment by Sammers21** [comment]
moved to https://github.com/vert-x3/vertx-cassandra-client/pull/4

**Pull request closed without merging**